src/dataflow/src/sink/kafka.rs

```diff
@@ -1012,15 +1019,24 @@ where
             // updates. Only on worker receives all the updates and we don't want
             // the other workers to also emit END records.
             if is_active_worker {
-                if let Err(e) = s.maybe_emit_progress(frontier.borrow()).await {
-                    // This can happen when the producer has not been
-                    // initialized yet. This also means, that we only start
-                    // emitting continuous updates once some real data
-                    // has been emitted.
-                    debug!("Error writing out progress update: {}", e);
+                match s.maybe_emit_progress(frontier.borrow()).await {
+                    Ok(progress_emitted) => {
+                        if progress_emitted {
```

Comment:
This change is necessary because unconditionally putting the flush on the hot path every time has a noticeable perf impact.  mz runs around 200 - 400% CPU for me locally after just running the `kafka-exactly-once-sinks` testdrive file